girder/models/file.py

```diff
@@ -222,19 +228,35 @@ def createFile(self, creator, item, name, size, assetstore, mimeType=None,
             'assetstoreId': assetstore['_id'],
             'name': name,
             'mimeType': mimeType,
-            'size': size
+            'size': size,
+            'itemId': item['_id'] if item else None
         }
 
-        if item:
-            file['itemId'] = item['_id']
-            self.propagateSizeChange(item, size)
-        else:
-            file['itemId'] = None
-
         if saveFile:
             return self.save(file)
         return file
 
+    def _propagateSizeToItem(self, event):
+        """
+        This callback updates an item's size to include that of a newly-created
+        file.
+
+        This generally should not be called or overridden directly. This should
+        not be unregistered, as that would cause item, folder, and collection
+        sizes to be inaccurate.
+        """
+        # This task is not performed in "createFile", in case
+        # "saveFile==False". The item size should be updated only when it's
+        # certain that the file will actually be saved. It is also possible for
+        # "model.file.save" to set "defaultPrevented", which would prevent the
+        # item from being saved initially.
+
+        fileDoc = event.info
+        itemId = fileDoc.get('itemId')
+        if itemId:
```

Comment:
`if itemId and fileDoc.get('size', 0):`

Link files don't have a size field, and we shouldn't call the method for 0-length files anyway.